addons/sourcemod/scripting/superzombiefortress.sp

```diff
@@ -1445,6 +1527,21 @@ void EndGracePeriod()
 	zf_spawnSurvivorsLastDeath = GetGameTime();
 }
 
+public void Frame_PostGracePeriodSpawn(int iClient)
+{
+	ChangeClientTeam(iClient, zomTeam());
+	
+	if (!IsPlayerAlive(iClient))
+	{
+		if (zomTeam() == INT(TFTeam_Blue))
+			ShowVGUIPanel(iClient, "class_blue");
+		else
+			ShowVGUIPanel(iClient, "class_red");
+	}
+						
+	g_bWaitingForTeamSwitch[iClient] = false;
```

Comment:
Best to put this in round start client loop, outside of valid client check. Incase client leaves while having `g_bWaitingForTeamSwitch` set to true

Reply:
Did everything but for this case I just set `g_bWaitingForTeamSwitch` to false when clients disconnect, because grace period can end in cases other than end of setup